modules/publish/controllers/Main.php

```diff
@@ -399,21 +399,37 @@ public function runFindArticles(framework\Request $request)
          */
         public function runToggleFavouriteArticle(framework\Request $request)
         {
-            if ($article_id = $request['article_id'])
+            // Read request parameters.
+            $article_id = $request['article_id'];
+            $user_id = $request['user_id'];
+
+            // Validate request parameters.
+            if ($article_id === null)
             {
-                try
-                {
-                    $article = Articles::getTable()->selectById($article_id);
-                    $user = \thebuggenie\core\entities\User::getB2DBTable()->selectById($request['user_id']);
-                }
-                catch (\Exception $e)
-                {
-                    return $this->renderText('fail');
-                }
+                return $this->renderText('no article');
```

Comment:
`return404()`

Reply:
This one should probably be a `400`, since the request did not include correct parameter?